### DISCUSSION TITLE
@Reflective and ReflectiveProcessor support for KafkaListener

### DISCUSSION THREAD

**Discussion by MatejNedic** [spring-projects/spring-kafka#2765] [Ideas] [answered]
Hey,

I am working on adapting Native hints for Spring Cloud AWS SQS and since I am looking into Spring Kafka for inspiration I noticed `@KafkaListener` is not annotated with `@Reflective` and is not marking parameters for reflection.
Since I am not that to deep into Spring Kafka internals I am not sure if this is needed or you are doing it some other way. If it is something you think is good and would be useful to a project I am willing to make a PR!

**Comment by garyrussell** [comment, marked as answer]
See 

https://github.com/spring-projects/spring-kafka/blob/08e76d3364f02f8161a9bbd7532d485d231acb0c/spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaRuntimeHints.java#L89-L96

We did add some code to detect AVRO types in listener methods, but everything works without `@Reflective` on `@KafkaListener`.

https://github.com/spring-projects/spring-kafka/blob/08e76d3364f02f8161a9bbd7532d485d231acb0c/spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaAvroBeanRegistrationAotProcessor.java#L38-L46

**Reply by MatejNedic** [reply]
Ohhhh didn't see that part for Avro. Sorry for raising a false flag then 😅.

I have seen a RuntimeHintsRegistrar, but without checking internals of Spring Kafka and Spring Messaging it is hard to know really what hints represent. We had issue where class that has `@SqsListener` needed to be marked for reflection and Class which is Parameter of a method as well since Mapping2Jackson would fail otherwise. 
Thanks on quick answer.

**Comment by artembilan** [comment]
The `@KafkaListener` is marked as `@MessageMapping`, which, in turn, is `@Reflective(MessageMappingReflectiveProcessor.class)`.
but I see that `@SqsListener` is not:
```
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface SqsListener {
```

So, that's an answer for you how it work in Spring for Apache Kafka.

**Reply by garyrussell** [reply]
Thanks, @artembilan I had forgotten about `@MessageMapping`; it's been there since day 1.

**Reply by MatejNedic** [reply]
Have missed that annotation. Thanks. I have created my own Processor in PR for native hints, but will replace it with `@MessageMappin` tnx @artembilan
